src/coreclr/jit/emitarm64sve.cpp

```diff
@@ -18421,4 +18421,243 @@ void emitter::getInsSveExecutionCharacteristics(instrDesc* id, insExecutionChara
+        case IF_SVE_AA_3A: // <Zdn>.<T>, <Pg>/M, <Zdn>.<T>, <Zm>.<T>
+        case IF_SVE_AB_3B: // <Zdn>.D, <Pg>/M, <Zdn>.D, <Zm>.D
+        case IF_SVE_AC_3A: // <Zdn>.<T>, <Pg>/M, <Zdn>.<T>, <Zm>.<T>
+        case IF_SVE_AO_3A: // <Zdn>.<T>, <Pg>/M, <Zdn>.<T>, <Zm>.D
+        case IF_SVE_CM_3A: // <Zdn>.<T>, <Pg>, <Zdn>.<T>, <Zm>.<T>
+        case IF_SVE_GP_3A: // <Zdn>.<T>, <Pg>/M, <Zdn>.<T>, <Zm>.<T>, <const>
+        case IF_SVE_GR_3A: // <Zdn>.<T>, <Pg>/M, <Zdn>.<T>, <Zm>.<T>
+        case IF_SVE_HL_3A: // <Zdn>.<T>, <Pg>/M, <Zdn>.<T>, <Zm>.<T>
+        case IF_SVE_HL_3B: // <Zdn>.H, <Pg>/M, <Zdn>.H, <Zm>.H
+            assert(movprefxIsPredicated);
+            assert(secondId->idReg1() != secondId->idReg4());
+            break;
+
+        // Tied registers
+        case IF_SVE_AV_3A: // <Zdn>.D, <Zdn>.D, <Zm>.D, <Zk>.D
+            assert(!movprefxIsPredicated);
+            assert(secondId->idReg1() != secondId->idReg3());
+            assert(secondId->idReg1() != secondId->idReg4());
+            break;
+
+        case IF_SVE_AR_4A: // <Zda>.<T>, <Pg>/M, <Zn>.<T>, <Zm>.<T>
+        case IF_SVE_AS_4A: // <Zdn>.<T>, <Pg>/M, <Zm>.<T>, <Za>.<T>
+        case IF_SVE_GT_4A: // <Zda>.<T>, <Pg>/M, <Zn>.<T>, <Zm>.<T>, <const>
+        case IF_SVE_HU_4A: // <Zda>.<T>, <Pg>/M, <Zn>.<T>, <Zm>.<T>
+        case IF_SVE_HU_4B: // <Zda>.H, <Pg>/M, <Zn>.H, <Zm>.H
+        case IF_SVE_HV_4A: // <Zdn>.<T>, <Pg>/M, <Zm>.<T>, <Za>.<T>
+            assert(movprefxIsPredicated);
+            assert(secondId->idReg1() != secondId->idReg3());
+            assert(secondId->idReg1() != secondId->idReg4());
+            break;
+
+        case IF_SVE_AT_3A: // <Zd>.<T>, <Zn>.<T>, <Zm>.<T>
+            // Only a subset of this group is valid
+            switch (secondId->idIns())
+            {
+                case INS_sve_sclamp:
+                case INS_sve_uclamp:
+                case INS_sve_eorbt:
+                case INS_sve_eortb:
+                case INS_sve_fclamp:
+                    break;
+                default:
+                    assert(!"Got unexpected instruction format within group after MOVPRFX");
+            }
+            assert(!movprefxIsPredicated);
+            assert(secondId->idReg1() != secondId->idReg2());
+            assert(secondId->idReg1() != secondId->idReg3());
+            break;
+
+        default:
+            assert(!"Got unexpected instruction format after MOVPRFX");
+            break;
+    }
+
+    if (firstId->idInsFmt() == IF_SVE_AH_3A)
+    {
+        // 3 operand version
+
+        // "The prefixed instruction must specify the same predicate register"
+        assert(isPredicateRegister(firstId->idReg2()));
+        assert(isPredicateRegister(secondId->idReg2()));
+        assert(firstId->idReg2() == secondId->idReg2());
+
+        // "and have the same maximum element size (ignoring a fixed 64-bit "wide vector" operand),"
+        assert(firstId->idInsOpt() == secondId->idInsOpt());
```

Comment:
can this be true for unpredicated version as well?

Reply:
The unpredicated `movprfx` doesn't care about element sizes, it is just `MOVPRFX <Zd>, <Zn>`.